toxcore/ping.c

```diff
@@ -87,10 +87,10 @@ int32_t ping_send_request(Ping *ping, IP_Port ipp, const uint8_t *public_key)
     crypto_memzero(shared_key, sizeof(shared_key));
 
     if (rc != PING_PLAIN_SIZE + CRYPTO_MAC_SIZE) {
-        return 1;
+        return;
     }
 
-    return sendpacket(dht_get_net(ping->dht), ipp, pk, sizeof(pk));
+    sendpacket(dht_get_net(ping->dht), ipp, pk, sizeof(pk));
```

Comment:
Isn't this now an unused return value? I thought this function intentionally returns the failure.

Reply:
It does, but the return value is never looked at. Should we log `sendpacket` errors, or should we have `ping_send_request` return errors and log them all? My reluctance for doing that is that it might fill the log with unnecessary spam.

Reply:
We could leave it alone, or we could log it at TRACE level if we almost don't care, but it might be useful to see it when debugging. Does it actually fail quite often? What makes it spam the logs?

Reply:
The only case it looks like it would normally fail is when `sendpacket` fails, which is already pretty sufficiently logged. I guess we could log all the failures in that function with TRACE.

Reply:
Ok, add a short comment saying that here, and add a `(void)` cast to tell the static analyser that we mean it.